app/src/main/java/org/inspire/breath/HeartBeat.java

```diff
@@ -0,0 +1,52 @@
+package org.inspire.breath;
+
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.widget.TextView;
+import static java.nio.file.StandardOpenOption.*;
+import java.nio.file.*;
+import java.io.*;
+
+public class HeartBeat extends AppCompatActivity {
+    TextView heartBeats;
+    WawReader reader = new WawReader();
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_heart_beat);
+
+
+        try {
+            InputStream is = getResources().openRawResource(R.raw.heartbeat2);
+
+            ByteArrayOutputStream buffer = new ByteArrayOutputStream();
+
+            int nRead;
+            byte[] data = new byte[16384];
+
+            while ((nRead = is.read(data, 0, data.length)) != -1) {
+                buffer.write(data, 0, nRead);
+            }
+
+            byte [] arr = buffer.toByteArray();
+
+
+            File tempFile = File.createTempFile("tmp", ".wav", null);
+            FileOutputStream fos = new FileOutputStream(tempFile);
+            fos.write(arr);
+            int cnt = reader.getResult(tempFile);
+            heartBeats = (TextView)findViewById(R.id.res_field);
+            heartBeats.setText(""+cnt/2*6); // mult by 6, because file 10 sec long
+        }
+        catch (Exception e){
+            System.out.println("no read");
```

Comment:
System.out.println doesn't give a good log of the error, could you return to the main tests page if the audio file fails to read, and give a toast to alert the user?